src/app/helpers/initUserSettings.ts

```diff
@@ -0,0 +1,104 @@
+import f from "lodash/fp";
+import { either } from "./functools";
+import {
+  UserSettingKeyFilter,
+  UserSettingKeyGlobal,
+  UserSettingKeyTable,
+  UserSettingValue
+} from "../types/userSettings";
+import actions from "../redux/actionCreators";
+import store from "../redux/store";
+
+/** LocalStorage -> "globalSettings" */
+type GlobalSettings = {
+  [Key in UserSettingKeyGlobal]: UserSettingValue<Key>;
+};
+
+type TableSettings = {
+  [Key in UserSettingKeyTable]: UserSettingValue<Key>;
+};
+
+type FilterSettings = {
+  rowsFilter: UserSettingValue<"rowsFilter">;
+};
+
+type TableView = {
+  default: TableSettings;
+};
+
+type FilterView = {
+  [name: string]: FilterSettings;
+};
+
+/** LocalStorage -> "tableViews" */
+type TableViews = {
+  [tableId: number]: TableView;
+  "*": FilterView;
+};
+
+function readLocalStorage(key: string) {
+  return either(localStorage)
+    .map(f.get(key))
+    .map(JSON.parse)
+    .getOrElse({});
+}
+
+/**
+ * initialize user settings from localStorage
+ */
+async function initUserSettings() {
+  const globalSettings: GlobalSettings = readLocalStorage("globalSettings");
+  const tableViews: TableViews = readLocalStorage("tableViews");
+
+  // global settings
+  for (const [key, value] of f.entries(globalSettings)) {
```

Comment:
Statt hier dreimal `n` Updates zu triggern sollte man eventuell über die Entries mappen und Thunks generieren, die man dann einmal gesammelt mit `batch` an Redux schickt. React macht automatisches Update-/Renderbatching erst ab Version 18.

Reply:
Habe das ganze mal in ein `batch` gewrapped.